services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersValidationSingleSwFlowSpec.groovy

```diff
@@ -0,0 +1,411 @@
+package org.openkilda.functionaltests.spec.switches
+
+import static org.junit.Assume.assumeTrue
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.SwitchHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.command.switches.DeleteRulesAction
+import org.openkilda.messaging.error.MessageError
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+
+import groovy.transform.Memoized
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+import spock.lang.Unroll
+
+@Narrative("""This test suite check the switch validate feature on a single flow switch.
+Description of fields:
+- missing - those meters/rules, which are NOT present on a switch, but are present in db
+- misconfigured - those meters which have different value (on a switch and in db) for the same parameter
+- excess - those meters/rules, which are present on a switch, but are NOT present in db
+- proper - meters/rules values are the same on a switch and in db
+""")
+class MetersValidationSingleSwFlowSpec extends BaseSpecification {
+    @Autowired
+    SwitchHelper switchHelper
+
+    @Unroll
+    def "Switch validation is able to store correct information on a #switchType switch in the proper section"() {
+        assumeTrue("Unable to find required switches in topology", switches as boolean)
+
+        setup: "Select a #switchType switch and retrieve default meters"
+        def sw = switches.first()
+        def defaultMeters = northbound.getAllMeters(sw.dpId)
+
+        when: "Create a flow"
+        def flow = flowHelper.addFlow(flowHelper.singleSwitchFlow(sw))
+        def meterIds = getCreatedMetersId(sw.dpId, defaultMeters)
+        def burstSize = switchHelper.getExpectedBurst(sw.dpId, flow.maximumBandwidth)
+
+        then: "Two meters are automatically created."
+        meterIds.size() == 2
+
+        and: "The correct info is stored in the proper section"
+        def switchValidateInfo = northbound.switchValidate(sw.dpId)
+        switchValidateInfo.meters.proper.collect { it.meterId }.containsAll(meterIds)
+
+        def createdCookies = getCreatedCookies(sw.dpId, meterIds)
```

Comment:
This is non-sense. Why you pass `meterIds` to get created cookies? You can understand whether cookie is default or not by doing `org.openkilda.model.Cookie.isDefaultRule(cookie)` and get non-default cookies by doing
`northbound.getSwitchRules(switchId).flowEntries.findAll {  !Cookie.isDefault(it) }`

Reply:
method renamed, and now we can pass just SwitchId as parameter
getCreatedCookies -> getCookiesWithMeter

Reply:
the mentioned method was renamed and rewritten
fixed